src/main/kotlin/org/taktik/icure/utils/entities/embed/TimeTableItem.kt

```diff
@@ -104,26 +105,15 @@ fun TimeTableItem.iterator(startDateAndTime: Long, endDateAndTime: Long, duratio
 		}
 }
 
-fun List<TimeTableHour>.iterator(duration: Duration) = object : Iterator<Long> {
-	var current: LocalTime? = this@iterator.firstOrNull()?.let { (it.startHour ?: 0L).toLocalTime() } ?: throw NoSuchElementException()
-	var idx = 0
-
-	init {
-		if (!hasNext()) throw NoSuchElementException()
-	}
+fun List<TimeTableHour>.iterator(duration: Duration): Iterator<Long> = this.map { it.iterator(duration) }.sortedMerge()
+fun TimeTableHour.iterator(duration: Duration) = object : Iterator<Long> {
+	var current: LocalTime? = (this@iterator.startHour ?: 0L).takeIf { (this@iterator.endHour ?: 0L) == 0L || it < (this@iterator.endHour ?: 240000) }?.toLocalTime()
```

Comment:
Should you not check also for enough time (according to duration) between the start and end hour?